app/helpers/application_helper.rb

```diff
@@ -150,6 +164,26 @@ def current_user_from_header
     nil
   end
 
+  # Allow a user to log in as a person if they know the uuid of an opened response.
+  def current_user_from_request_path
+    # Since this potentially has security implications, make it a feature toggle. Defaults to false.
+    return nil unless Rails.application.config.settings.feature_toggles.allow_response_uuid_login
+
+    # Works on both the questionnaire and the questionnaire/preference urls
+    uuid_match = %r{\A/questionnaire/(?:preference/)?([a-z0-9-]+)/?\z}.match(request&.path)
+    return nil unless uuid_match.present?
+
+    # The response must be opened and cannot be completed already
+    # (this is so that once a response has been completed, that link cannot
+    #  be used to log in for that same user again)
+    response = Response.opened.find_by(uuid: uuid_match[1])
```

Comment:
yes because the `(?:)` introduces a regex backreference that is not included in the match list (as opposed to a `()` match that is included). I'll add a comment.